include/RandLAPACK/comps/qb.hh

```diff
@@ -0,0 +1,105 @@
+#ifndef BLAS_HH
+#include <blas.hh>
+#define BLAS_HH
+#endif
+
+#include "orth.hh"
+#include "rf.hh"
+
+namespace RandLAPACK::comps::qb {
+
+#ifndef QB_CLASS
+#define QB_CLASS
+
+template <typename T>
+class QBalg
+{
+        public:
+                virtual void call(
+                        int64_t m,
+                        int64_t n,
+                        std::vector<T>& A,
+                        int64_t& k,
+                        int64_t block_sz,
+                        T tol,
+                        std::vector<T>& Q,
+                        std::vector<T>& B
+                ) = 0;
+};
+
+template <typename T>
+class QB : public QBalg<T>
+{
+	public:
+                RandLAPACK::comps::rf::RangeFinder<T>& RF_Obj;
+                RandLAPACK::comps::orth::Stabilization<T>& Orth_Obj;
+                bool verbosity;
+                bool orth_check;
+
+                // Avoiding preallocations
```

Comment:
I don't understand what this comment is getting at. Based on what I see, I'd call this "declare (but do not allocate) workspace".